finagle-base-http/src/main/scala/com/twitter/finagle/http/headers/JMapBackedHeaderMap.scala

```diff
@@ -0,0 +1,99 @@
+package com.twitter.finagle.http.headers
+
+import com.twitter.finagle.http.HeaderMap
+import java.util.function.BiConsumer
+import scala.collection.AbstractIterator
+
+/**
+ * Mutable, thread-safe [[HeaderMap]] implementation, backed by
+ * a mutable [[Map[String, Header]]], where the map key
+ * is forced to lower case
+ */
+private[http] trait JMapBackedHeaderMap extends HeaderMap {
```

Comment:
This feels like it should be an `abstract class`, and probably just vanilla `private`.